### PR TITLE
Add support for Vernier profiler

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Needs Ruby >=3.2 so blocked on that.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/prof.rb [hunk 1]
```diff
@@ -39,11 +41,15 @@ def run
         Homebrew.setup_gem_environment!
 
         if args.stackprof?
+          # odeprecated
+          # no longer maintained upstream
```
Comment (Bo98) [inline Library/Homebrew/dev-cmd/prof.rb:45]: There was an update 4 months ago?

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/prof.rb]: Reworded comment.

**Comment by carlocab** [comment]
How slow/fast is this meant to be? I've been staring at this for the past 5 minutes or so:
```
❯ brew prof --vernier -- deps --tree --include-build --include-test --annotate ffmpeg
Fetching gem metadata from https://rubygems.org/.......
Fetching ruby-prof 1.7.0
Fetching sorbet-runtime 0.5.11383
Fetching stackprof 0.2.26
Fetching vernier 1.0.1
Installing vernier 1.0.1 with native extensions
Installing stackprof 0.2.26 with native extensions
Installing sorbet-runtime 0.5.11383
Installing ruby-prof 1.7.0 with native extensions
Bundle complete! 38 Gemfile dependencies, 40 gems now installed.
Bundled gems are installed into `../../../../../../../opt/homebrew/Library/Homebrew/vendor/bundle`
starting profiler with interval 500
```

**Review by carlocab — APPROVED** [review]
I haven't gotten this to work, but it doesn't seem to break anything, so fine with me.

**Comment by MikeMcQuaid** [comment]
@carlocab Thanks! Let's merge and iterate. Hopefully can help you debug this 🔜

**Comment by MikeMcQuaid** [comment]
> I've been staring at this for the past 5 minutes or so:

I saw it hang a few times, unsure why. It worked if I Ctrl-Cd and tried again.

Probably worth us figuring this out as it's _way_ better than ruby-prof etc.

**Comment by carlocab** [comment]
> I saw it hang a few times, unsure why. It worked if I Ctrl-Cd and tried again.

Did this several times and didn't get anywhere, sadly.

**Pull request merged**